File: backend/app/models/friendlink.py
"""Friend links domain model."""

from __future__ import annotations

from datetime import UTC, datetime, timedelta
from typing import Annotated

from beanie import Document, Indexed
from pydantic import Field


class FriendLinks(Document):
    """友情链接文档模型"""

    name: str
    url: str
    sort_order: Annotated[int, Indexed()] = 0
    favicon: str | None = None
    description: str | None = None
    email: str | None = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))

    class Settings:
        name = "friend_links"
        use_cache = True
        cache_expiration_time = timedelta(seconds=600)

File: go-backend/internal/mongo/document/friend_link.go
package document

import "time"

type FriendLink struct {
	ID          string    `bson:"_id,omitempty"`
	Name        string    `bson:"name"`
	URL         string    `bson:"url"`
	SortOrder   int       `bson:"sort_order"`
	Favicon     *string   `bson:"favicon"`
	Description *string   `bson:"description"`
	Email       *string   `bson:"email"`
	CreatedAt   time.Time `bson:"created_at"`
}
